### ISSUE TITLE
Criteria NEQ produces EQ

### ISSUE THREAD

**Issue by flipspl** [doctrine/doctrine2#6036]
I have this Criteria in my entity to get all the non removed sibling entities from the same server:

```
$serverCriteria = Criteria::create()->where(Criteria::expr()->andX(
    Criteria::expr()->eq('server', $entity->getServer()),
    Criteria::expr()->eq('is_removed', 0),
    Criteria::expr()->neq('id', $entity->getId())
));
```

But the query that is produced instead of being

```
... AND et.id <> 50;
```

is:

```
... AND et.id = 50;
```

**Comment by jaikdean** [comment]
The same issue I have in #5687

**Comment by lcobucci** [comment]
@flip111 @jaikdean can you abstract it into a test case? See https://github.com/doctrine/doctrine2/tree/d3f6c5ec70aac4b029a4b61ecf1e2ba61a1a4a6d/tests/Doctrine/Tests/ORM/Functional/Ticket for examples.

**Comment by flip111** [comment]
i feel honoured by you mention!

**Comment by lcobucci** [comment]
@flip111 sorry (autocomplete hell) But you know, if you want to collaborate here's a good opportunity HAHAHAHA

**Comment by flip111** [comment]
didn't run this code, but could be something like this

```php
    // in: Doctrine\Tests\ORM\Persisters\Collection\ManyToManyPersisterTest
    public function testInvokeLoadCriteria()
    {
        $idToFilter = 2;

        $owning = new \Doctrine\Tests\Models\ValueConversionType\OwningManyToManyEntity();
        $inversed1 = new \Doctrine\Tests\Models\ValueConversionType\InversedManyToManyEntity();
        $inversed2 = new \Doctrine\Tests\Models\ValueConversionType\InversedManyToManyEntity();

        // Create the collection as if it's just been queried from the database.
        $inv1 = clone $inversed1;
        $inv1->id1 = 1;

        $inv2 = clone $inversed2;
        $inv2->id1 = $idToFilter;

        $collection = Doctrine\ORM\PersistentCollection(
            $this->em,
            $this->em->getClassMetadata('Doctrine\Tests\Models\ValueConversionType\InversedManyToManyEntity'),
            new Doctrine\Common\Collections\ArrayCollection\ArrayCollection([$inv1, $inv2]);
        );

        // Set the actual "collection" (this should become a PersistentCollection after having been queried)
        $this->owning = [$inversed1, $inversed2]; // trusting doctrine knows how to store an array in the right table

        // Now need a test database (sqlite probably)
        // todo: store $owning, $inversed1 and $inversed2

        // Try to match the Criteria from the PersistentCollection
        $criteria = Criteria::create()->where(
            Criteria::expr()->neq('id1', $idToFilter)
        );

        // Get the new collection
        $filtered = $owning->associatedEntities()->matching($criteria);

        // Test to see if anything has been filtered.
        $this->assertNotEquals($collection, $filtered); // could do more sophisticated tests here ..
    }
```

**Comment by jaikdean** [comment]
I think this is fixed by #6740

**Comment by lcobucci** [comment]
@jaikdean thanks! closing then 👍
